### PR TITLE
First attempt in adding NumPy style docstrings

### PR DESCRIPTION
**Pull request by ysuarez**
## Link to Github issue or other discussion

#220 

## What does this PR do?

This PR adds NumPy style docstrings to a couple of functions.

## What changes were made?

I reformatted pre-existing comments summarizing what a ew functions accomplished, then added some NumPy style function docstrings that matched the style that the code base was already using.

Here is an example of this style...

``
def set_media_type(config, filepath, file_fieldname, csv_row):
    """Using either the 'media_type' or 'media_types_override' configuration
       setting, determine which media bundle type to use.

       Parameters
       ----------
       config : dict
           The configuration settings defined by workbench_config.get_config().
       filepath: string
           The value of the CSV 'file' column.
       file_fieldname: string
            The name of the CSV column containing the filename (usually 'file').
       csv_row : OrderedDict
            The CSV row for the current item.
       Returns
       -------
       string
           A string naming the configured media type, e.g. 'image'.
    """
<snip>
```

## How to test / verify this PR?

This needs to be tested by comparing the details of the new docstrings with the actual function code, params, and return value(s).

## Interested Parties

@mjordan

---

## Checklist

* [x] Before opening this PR, have you opened an issue explaining what you want to to do?
* [x] Have you run `pycodestyle --show-source --show-pep8 --ignore=E402,W504 --max-line-length=200 yourfile.py`? 
* [ ] Have you included some configuration and/or CSV files useful for testing this PR?
* [ ] Have you written unit or integration tests if applicable?
* [ ] Does the code added in this PR require a version of Python that is higher than the current minimum version?
* [ ] If the changes in this PR require an additional Python library, have you included it in `setup.py`?
* [ ] If the changes in this PR add a new configuration option, have you provided a default for when the option is not present in the .yml file?

### CODE REVIEW

**Comment by ysuarez** [comment]
@mjordan I am not sure how detailed I should get in explaining what the parameters do. Of course my interpretation of the code could be wrong, and maybe simpler / higher level explanations are better?

**Comment by ysuarez** [comment]
I have an update, after looking at more of the code in workbench_utils.

* I am not sure how you want me to handle indentation

- current style, first line next to the 3 quotes, and the rest of the content indented a total of 8 spaces

```
    """Using either the 'media_type' or 'media_types_override' configuration
       setting, determine which media bundle type to use.

       Parameters
       ----------
    """

- versus:  first line next to the 3 quotes, and the rest of the content indented a total of 4 spaces

```
    """Using either the 'media_type' or 'media_types_override' configuration
    setting, determine which media bundle type to use.

    Parameters
    ----------
    """
- versus:  first line next after the 3 quotes, and the all of the content indented a total of 4 spaces

```
    """
    Using either the 'media_type' or 'media_types_override' configuration
    setting, determine which media bundle type to use.

    Parameters
    ----------
    """

**Comment by ysuarez** [comment]
@mjordan let me know if you any issues with the docstrings I am creating, or the existing ones I am updating. For now I am only adding docstrings to workbench_utils.py.

I prefer to get some confirmation that you approve if my general approach so far before I get too deep in making changes to his first PR for issue #220.

**Comment by mjordan** [comment]
@ysuarez re. indentation of comment strings, I'm not fussy other than it should pass pycodestyle.

**Comment by ysuarez** [comment]
@mjordan yes, the "current style" passes pycodestyle and Pycharm checks. I will go back and make all the docstrings match the "current style."

BTW, one thing I found is that in at least one instance, a single parameter had 9 spaces (one too many, while everything else used 8 spaces) and Pycharm warned me to fix it.

Any comments on how I am describing the params and returns? Am I doing too much explanations of each param/return, not enough?

**Comment by ysuarez** [comment]
Also, not sure how to best proceed to not trigger lots of Github notifications as I work. 

For example, should I put in a final set of commits to this PR and we ask you to merge it (assuming you like it). Then I can start a new PR and keep it in draft mode, or just have a topic branch and when I reach a certain threshold make a new PR then?

I just want to create the least number of Github notifications for you, except when I do need your feedback.

File: workbench_utils.py [hunk 1]
```diff
@@ -131,7 +132,21 @@ def get_oembed_url_media_type(config, filepath):
...
+
+    Parameters
+    ----------
+    config : dict
+           The configuration settings defined by workbench_config.get_config().
+
+    Returns
+    -------
+    media_types : list
+        A list with the configured allowed oEmbed media type(s), e.g. 'remote_video'.
```
Comment (mjordan) [inline workbench_utils.py:146]: If the return value is a list, I'd prefer it to read `e.g. '[remote_video']`.

**Comment by ysuarez** [comment]
@mjordan I think you can try to merge now.

Though I am not totally  happy with the docstring for `def get_entity_fields(config, entity_type, bundle_type):`

https://github.com/ysuarez/islandora_workbench/blob/92b770e76669e9212b1954c493c2b73c3cd7a189/workbench_utils.py#L1161-L1178

Once this is merged, I will start a new topic branch, then a 2nd PR.

**Comment by ysuarez** [comment]
@mjordan would you like me to turn this PR into smaller PRs to make it easier to review and to merge? For example, I can break it into like 3 PRs.

**Comment by mjordan** [comment]
Thanks @ysuarez !

**Pull request merged**